content/advent-2017/kubernetes-ready-service.md

```diff
@@ -0,0 +1,763 @@
+  rules:
+  - host: advent.test
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: {{ .ServiceName }}
+          servicePort: 80
+```
+
+Now to check how it works we need to install and run `minikube`, its official documentation is [here](https://github.com/kubernetes/minikube#installation). We also need the [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) tool to be able to apply the configuration and check the service.
+
+To start `minikube`, enable ingress and prepare `kubectl` we need to run few commands:
+
+```bash
+minikube start
+minikube addons enable ingress
+kubectl config use-context minikube
+```
+
+Now let's add a new `Makefile` goal to be able to install the service on `minikube`:
+
+`Makefile`
+```make
+minikube: push
+	for t in $(shell find ./kubernetes/advent -type f -name "*.yaml"); do \
+        cat $$t | \
+        	gsed -E "s/\{\{(\s*)\.Release(\s*)\}\}/$(RELEASE)/g" | \
+        	gsed -E "s/\{\{(\s*)\.ServiceName(\s*)\}\}/$(APP)/g"; \
+        echo ---; \
+    done > tmp.yaml
+	kubectl apply -f tmp.yaml
+```
+
+These commands "compile" all `*.yaml` configurations to a single file, replace `Release` and `ServiceName` "variables" by the real values (please, note that here I use `gsed` instead of the standard `sed`) and run `kubectl apply` to install the application to Kubernetes.
+
+Let's check if our configuration works:
+
+```bash
+$ kubectl get deployment
+NAME      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+advent    3         3         3            3           1d
+
+$ kubectl get service
+NAME         CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+advent       10.109.133.147   <none>        80/TCP    1d
+
+$ kubectl get ingress
+NAME      HOSTS         ADDRESS        PORTS     AGE
+advent    advent.test   192.168.64.2   80        1d
+```
+
+Now we can try to make requests to the service. But first of all, we need to add our fake domain `advent.test` to the `/etc/host` file:
+
+```bash
+echo "$(minikube ip) advent.test" | sudo tee -a /etc/hosts
+```
+
+And now finally we can check our service:
+
+```bash
+curl -i http://advent.test/home
+HTTP/1.1 200 OK
+Server: nginx/1.13.6
+Date: Sun, 10 Dec 2017 20:40:37 GMT
+Content-Type: application/json
+Content-Length: 72
+Connection: keep-alive
+Vary: Accept-Encoding
+
+{"buildTime":"2017-12-10_11:29:59","commit":"020a181","release":"0.0.5"}%
+```
+
+Yeah, it works!
+
+---
+
+You can find all steps [here](https://github.com/rumyantseva/advent-2017), there are two versions available: [commit-by-commit](https://github.com/rumyantseva/advent-2017/commits/master) and [all steps in one](https://github.com/rumyantseva/advent-2017/tree/all-steps).
+
+If you like this article, it might be interesting how may look like a more flexible service, prepared for the real prodution. In this case, feel free to take a look at [takama/k8sapp](https://github.com/takama/k8sapp), a Go application template which meets the Kubernetes requirements.
```

Comment:
what do you think about `it might be interesting for you how a more flexible service, prepared for the real production, may look like` for summary?